### ISSUE TITLE
What's the inspiration behind the supervised loss function?

### ISSUE THREAD

**Issue by bongbonglemon** [senguptaumd/Background-Matting#15]

**Comment by senguptaumd** [comment]
Can you tell me what you mean by inspiration? The supervised loss was very similar to previous works, nothing new. Gradient loss was used by Late Fusion matting, it helps to get sharper edges in the alpha, else the network is biased towards smoother edges

**Comment by bongbonglemon** [comment]
I looked into the Late Fusion matting paper but I could not find the loss that you used so I thought your loss was an original one and wanted to know how you made it.

**Comment by senguptaumd** [comment]
The supervised L1 loss is similar to Deep Image Matting paper (they have losses over alpha and composition). We did not attempt to make it differentiable and use simple L1 loss, also add a loss over the foreground F. Note that many existing matting works do not predict the foreground layer F and simply use the image I while compositing. This causes artifacts (see our paper appendix).

The gradient loss is added from Late Fusion Matting paper (eq 5)

**Comment by bongbonglemon** [comment]
So you are saying this loss (from your paper):
![Screen Shot 2020-04-11 at 12 57 29 AM](https://user-images.githubusercontent.com/40830439/79008334-bd9dda00-7b8f-11ea-89f4-31a748260fb0.png)
is similar to these losses (from the DIM paper):
![Screen Shot 2020-04-11 at 1 00 39 AM](https://user-images.githubusercontent.com/40830439/79008401-e8882e00-7b8f-11ea-994c-2113a21145f8.png)

and
![Screen Shot 2020-04-11 at 1 00 45 AM](https://user-images.githubusercontent.com/40830439/79008410-ecb44b80-7b8f-11ea-96a0-0bee363f47f5.png)
And the term with the gradient is from the Late Fusion Matting paper:
![Screen Shot 2020-04-11 at 1 02 18 AM](https://user-images.githubusercontent.com/40830439/79008505-1d948080-7b90-11ea-815a-6b70ebb88406.png)
Thank you for your time and patience. Just want to be sure. I'm admittedly not good at this math. Its just that these equation don't seem to match...at least on vision inspection.

**Comment by senguptaumd** [comment]
In our paper loss of alpha, F, and the last term is L1 loss (absolute distances). In the DIM paper it is L2 loss. DIM shows how to make the gradient differentiable. This is not needed anymore as pytorch can handle it through auto-diff. Loss in eq5 of LFM is same as the 2nd term in our paper.

**Comment by bongbonglemon** [comment]
Thank you so much! This has brought alot of clarity! One last question... why does the loss of F get more weight/coefficient of 2.

**Comment by senguptaumd** [comment]
Empirically I saw that to be effective. One explanation, learning matte is easier than the foreground, as alpha matte is mostly 0 or 1 and continuous at only certain regions. Often these weights are chosen empirically to make sure all loss components are converging at the same rate.
